mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -89,12 +93,30 @@ impl S3CrtClient {
 
         let mut client_bootstrap = ClientBootstrap::new(&allocator, &bootstrap_options).unwrap();
 
-        let creds_options = CredentialsProviderChainDefaultOptions {
-            bootstrap: &mut client_bootstrap,
-        };
+        let mut client_config = ClientConfig::new();
 
-        let mut creds_provider = CredentialsProvider::new_chain_default(&allocator, &creds_options).unwrap();
-        let signing_config = init_default_signing_config(region, &mut creds_provider);
+        if !config.no_sign_request {
+            let credentials_provider = match config.profile_name_override {
+                Some(profile_name_override) => {
+                    let credentials_profile_options = CredentialsProviderProfileOptions {
+                        bootstrap: &mut client_bootstrap,
+                        profile_name_override: Some(&profile_name_override),
+                    };
+                    CredentialsProvider::new_profile(&allocator, credentials_profile_options)
+                        .map_err(|_| CredentialsError::ProfileNotFound)
```

Comment:
I don't think `ProfileNotFound` is an accurate description of this error. Looking at https://github.com/awslabs/aws-c-auth/blob/main/source/credentials_provider_profile.c#L366, it could be:
* an invalid path to a profile file
* an invalid path to a credentials file
* the profile wasn't specified
* the profile doesn't exist
* or an allocation failure

So maybe I'd call this a more generic `CredentialsError::ProviderFailure`.

We also want it to not throw away the underlying error, so the enum variant should have a `#[source] mountpoint_s3::common::error::Error` field.